packages/core/src/registry/StyleSheetRegistry.ts

```diff
@@ -7,93 +8,105 @@ const STYLE_ID_PREFIX = "__";
 
 export class StyleSheetRegistry {
   private sheet: StyleSheet;
-  private serverSideRenderedStyles: Record<string, HTMLStyleElement> | null =
-    null;
-  private indices: Record<string, number | undefined> = {};
-  private instancesCounts: Record<string, number | undefined> = {};
+  private serverSideRenderedStyleMap: {
+    [id: string]: HTMLStyleElement;
+  } | null = null;
+  private indexesMap: { [id: string]: number[] | undefined } = {};
+  private instancesCountMap: { [id: string]: number | undefined } = {};
 
   constructor() {
     this.sheet = new StyleSheet("kuma-ui", isProduction);
     this.sheet.inject();
   }
 
-  public add(id: string, rule: string): void {
-    if (isBrowser && this.serverSideRenderedStyles === null) {
-      this.serverSideRenderedStyles = this.getServerSideRenderedStyles();
-      Object.keys(this.serverSideRenderedStyles).forEach((id) => {
-        this.instancesCounts[id] = 0;
-        this.indices[id] = -1;
+  public add(id: string, css: string): void {
+    if (isBrowser && this.serverSideRenderedStyleMap === null) {
+      this.serverSideRenderedStyleMap = this.getServerSideRenderedStyleMap();
+      Object.keys(this.serverSideRenderedStyleMap).forEach((id) => {
+        this.instancesCountMap[id] = 0;
       });
     }
 
-    this.instancesCounts[id] = 1 + (this.instancesCounts[id] ?? 0);
-    const serverSideRenderedStyle = this.serverSideRenderedStyles?.[id];
-    if (this.instancesCounts[id] === 1 && !serverSideRenderedStyle) {
-      this.indices[id] = this.sheet.insertRule(rule, this.indices[id]);
+    this.instancesCountMap[id] = 1 + (this.instancesCountMap[id] ?? 0);
+    const serverSideRenderedStyle = this.serverSideRenderedStyleMap?.[id];
+    if (this.instancesCountMap[id] === 1 && !serverSideRenderedStyle) {
+      compile(css).forEach((element) => {
+        const rule = serialize([element], stringify);
+        this.indexesMap[id] = (this.indexesMap[id] || []).concat(
+          this.sheet.insertRule(rule)
+        );
+      });
```

Comment:
fixed to process multiple rule strings using `stylis` and insert each rule one by one.